pkg/analyze/kube_resource.go

```diff
@@ -0,0 +1,172 @@
+package analyzer
+
+import (
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"strconv"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	iutils "github.com/replicatedhq/troubleshoot/pkg/interfaceutils"
+	"gopkg.in/yaml.v2"
+)
+
+func FindResource(kind string, namespace string, name string, getFileContents func(string) (map[string][]byte, error)) (interface{}, error) {
+
+	filemap := map[string]string{
+		"Deployment":           "deployments",
+		"StatefulSet":          "statefulsets",
+		"NetworkPolicy":        "network-policy",
+		"Pod":                  "pods",
+		"Ingress":              "ingress",
+		"Service":              "services",
+		"ResourceQuota":        "resource-quotas",
+		"Job":                  "jobs",
+		"PersistentVoumeClaim": "pvcs",
+		"pvc":                  "pvcs",
+		"ReplicaSet":           "replicasets",
+		"Namespace":            "namespaces.json",
+		"PersistentVolume":     "pvs.json",
+		"pv":                   "pvs.json",
+		"Node":                 "nodes.json",
+		"StorageClass":         "storage-classes.json",
+	}
+
+	var datapath string
+
+	resourceLocation, ok := filemap[kind]
+
+	if ok != true {
+		return nil, errors.New("failed to find resource")
+	}
+
+	if namespace != "" {
+		datapath = filepath.Join("cluster-resources", resourceLocation, fmt.Sprintf("%s.json", namespace))
+	} else {
+		datapath = filepath.Join("cluster-resources", resourceLocation)
+	}
+
+	files, err := getFileContents(datapath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read collected resources")
+	}
+
+	var resource interface{}
+	for _, file := range files {
+		err = yaml.Unmarshal(file, &resource)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse data as yaml doc")
+		}
+	}
+
+	var selected interface{}
+	items, err := iutils.GetAtPath(resource, "items")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get items from file")
+	}
+
+	itemslice := items.([]interface{})
+	for _, item := range itemslice {
+		name, err := iutils.GetAtPath(item, "metadata.name")
+		if err != nil {
+			return nil, errors.Wrapf(err, "Failed to find resource with name: %s", name)
+		}
+		if name == name {
+			selected = item
+			break
+		}
+	}
+
+	return selected, nil
+
+}
+
+func analyzeResource(analyzer *troubleshootv1beta2.ClusterResource, getFileContents func(string) (map[string][]byte, error)) (*AnalyzeResult, error) {
+
+	selected, err := FindResource(analyzer.Kind, analyzer.Namespace, analyzer.Name, getFileContents)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find resource")
```

Comment:
Shouldn't we treat not finding a resource business as usual and have analysis fail and reported in `AnalyzeResult.Message`?